code/modules/mob/living/simple_animal/friendly/corgi.dm

```diff
@@ -74,13 +75,16 @@
 			spawn(0) // Separate process
 				stop_automated_movement = 1
 				var/failedsteps = 0
-				while(failedsteps <= 3)
-					if(!movement_target || src.Adjacent(movement_target) || get_dist(src, movement_target) >= 7)
-						failedsteps = 4
-					if(!step_towards(src,movement_target,1))
-						failedsteps++
-					sleep(6)
-
+				var/infinite_chase = loc && locate(/obj/machinery/power/treadmill) in loc
+				while(TRUE)
```

Comment:
why don't you just make this while statement the check below?

`while(movement_target && !Adjacent(movement_target) && get_dist(src,movement_target)<7 && corgi_status == FOOD_HUNTING && failed_steps <= 2)`

Reply:
Readability :man_shrugging:

Reply:
It's exactly the same check, while(TRUE) is a bad practice

Reply:
Changed